omc3/scripts/merge_kmod_results.py

```diff
@@ -0,0 +1,275 @@
+            f"Computing lumi contribution from optics for IP {ip}"
+        )
+        ip_rows = data_frame.loc[data_frame.index.str.startswith(ip)]
+
+        beta_sums = [up.uarray(ip_rows[f"{BETA}{plane}"].to_numpy(),
+                               ip_rows[f"{ERR}{BETA}{plane}"].to_numpy()
+                               ).sum()  # sum over beams
+                     for plane in PLANES]
+        lumi_coefficient[ip] = 0.5 * up.sqrt(np.prod(beta_sums))
+
+    imbalance = lumi_coefficient[IPS[0]] / lumi_coefficient[IPS[1]]
+    LOG.info(f'{"Luminosity Imbalance":22s}: {imbalance:s}')
+    LOG.info(f'{"Effective beta IP1":22s}: {lumi_coefficient[IPS[0]]:s}')
+    LOG.info(f'{"Effective beta IP5":22s}: {lumi_coefficient[IPS[1]]:s}')
+
+    return imbalance, lumi_coefficient[IPS[0]], lumi_coefficient[IPS[1]]
+
+
+def _add_imbalance_to_header(tfs_df: tfs.TfsDataFrame,
+                             imbalance: UFloat, beta_ip1: UFloat, beta_ip5: UFloat) \
+        -> tfs.TfsDataFrame:
+    """
+    Function to add the calculated imablance and effective betas to the header
+
+    Args:
+        tfs_df (tfs.TfsDataFrame): a `TfsDataFrame` with the results from a kmod analysis.
+        imbalance (UFloat): uncertain imbalance
+        beta_ip1 (UFloat): uncertain effective beta in ip1
+        beta_ip5 (UFloat): uncertain effective beta in ip5
+
+    Returns:
+        tfs.TfsDataFrame with added headers.
+
+    """
+    header_map = [(HEADER_IMBALANCE, HEADER_REL_ERROR, imbalance),
+                  (HEADER_EFF_BETA_IP1, HEADER_REL_ERROR_IP1, beta_ip1),
+                  (HEADER_EFF_BETA_IP5, HEADER_REL_ERROR_IP5, beta_ip5),
+    ]
+    for val_key, err_key, val in header_map:
+        tfs_df.headers[val_key], tfs_df.headers[err_key] = str(val).split('+/-')
+
+    return tfs_df
+
+
+def merge_tfs(directories: List[pathlib.Path], filename: str) -> tfs.TfsDataFrame:
+    """
+    Merge different kmod analysis results from a list of directories into a single `TfsDataFrame`.
+
+    Args:
+        directories (List[pathlib.Path]): list of PosixPath objects to directories holding TFS
+            files with the results of kmod analysis.
+        filename (str): name of the TFS files to look for in the provided directories
+
+    Returns:
+        A `TfsDataFrame` combining all the loaded files from the provided directories.
+    """
+    # Combine the data into one tfs
+    new_tfs = tfs.TfsDataFrame()
+    headers = OrderedDict()
+    for d in sorted(directories):
+        loaded_tfs = tfs.read_tfs(d / filename)
+        headers.update(loaded_tfs.headers)  # old headers are lost in `append`
+        new_tfs = new_tfs.append(loaded_tfs, ignore_index=True)
+    new_tfs.headers = headers
+    new_tfs = new_tfs.set_index(NAME)
+
+    # drop BPMWK and check tfs
+    new_tfs = new_tfs.loc[~new_tfs.index.str.startswith(BPMWK), :]
+    _check_tfs_sanity(new_tfs)
+    return new_tfs
+
+
+def get_ip_dir_names(kmod_dirs: List[pathlib.Path]) -> List[pathlib.Path]:
```

Comment:
There was a comment somewhere, what this method does. Maybe it should have a better name? kmod_results_dirs?